### PR TITLE
#2438 move nodeNames to settings for ZookeeperClusterInfo

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2438)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by saivirtue** [comment]
> We have to keep the compatibility for Ohara Manager, hence we have to add the nodeNames to json response of ZookeeperClusterInfo. (it means you have to add pair (nodeNames -> "xxx") to ZOOKEEPER_CLUSTER_INFO_JSON_FORMAT manually)

You mean [this line](https://github.com/oharastream/ohara/blob/81baeb572dacd286b00f1fe53eba0c3053e5b5c5/ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/ZookeeperApi.scala#L188)?

I had updated that in #2531

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry it

File: ohara-agent/src/main/scala/com/island/ohara/agent/ZookeeperCollie.scala [hunk 1]
```diff
@@ -124,7 +124,6 @@ trait ZookeeperCollie extends Collie[ZookeeperClusterInfo] {
                         .nodeNames(successfulContainers.map(_.nodeName).toSet)
```
Comment (chia7712) [inline ohara-agent/src/main/scala/com/island/ohara/agent/ZookeeperCollie.scala]: my bad :( 

the nodeNames should be configured by input Creation than the sum of running containers. I have filed a issue for it (#2557)

Comment (saivirtue) [reply ohara-agent/src/main/scala/com/island/ohara/agent/ZookeeperCollie.scala]: That's OK.
I think #2557 is related to #2549 ; they both have same problem.
Let me reference the relation.

**Comment by saivirtue** [comment]
The failed tests are not related to this issue, will merge later.

**Pull request merged**